app/collins/controllers/actions/ipmi/GetPoolsAction.scala

```diff
@@ -0,0 +1,86 @@
+package collins.controllers.actions.ipmi
+
+import scala.concurrent.Future
+
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json.JsArray
+import play.api.libs.json.JsNumber
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+
+import collins.controllers.ResponseData
+import collins.controllers.SecureController
+import collins.controllers.actions.RequestDataHolder
+import collins.controllers.actions.SecureAction
+import collins.controllers.actions.ipaddress.AddressActionHelper
+import collins.models.IpmiInfo
+import collins.models.Truthy
+import collins.models.shared.AddressPool
+import collins.util.security.SecuritySpecification
+
+// Get pools, all or just those in use
+case class GetPoolsAction(
+  allPools: Truthy,
+  spec: SecuritySpecification,
+  handler: SecureController
+) extends SecureAction(spec, handler) with AddressActionHelper {
+
+  case class ActionDataHolder(all: Boolean) extends RequestDataHolder
+
+  override def validate(): Validation = Right(ActionDataHolder(allPools.toBoolean))
+
+  override def execute(rd: RequestDataHolder) = Future {
+    rd match {
+      case ActionDataHolder(all) =>
+        val pools: Set[String] =
+          if (all) {
+            IpmiInfo.AddressConfig.map(_.poolNames).getOrElse(Set())
+          } else {
+            IpmiInfo.AddressConfig.map(_.poolNames).getOrElse(Set())
+          }
+        format(pools)
+    }
+  }
+
+  protected def format(pools: Set[String]) = {
+    // name, gateway, network, startAddress
+    val addressPools = toAddressPools(pools)
+    val seq = Seq(
+      "POOLS" -> toJsonList(addressPools)
+    )
+    val jsObject = JsObject(seq)
+    ResponseData(Status.Ok, jsObject)
+  }
+
+  protected def toJsonList(pools: Set[AddressPool]): JsArray = JsArray(
+    pools.toList.map { pool =>
+      val seq = Seq(
+        "NAME" -> JsString(convertPoolName(pool.name, true)),
+        "NETWORK" -> JsString(formatNetworkAddress(pool.network)),
+        "START_ADDRESS" -> JsString(pool.startAddress.getOrElse("Unspecified")),
+        "SPECIFIED_GATEWAY" -> JsString(pool.gateway.getOrElse("Unspecified")),
+        "GATEWAY" -> JsString(pool.gateway.getOrElse(pool.ipCalc.minAddress)),
+        "BROADCAST" -> JsString(pool.ipCalc.broadcastAddress),
+        "POSSIBLE_ADDRESSES" -> JsNumber(pool.ipCalc.addressCount)
+      )
+      JsObject(seq)
+    }
+  )
+
+  protected def formatNetworkAddress(network: String): String = network match {
```

Comment:
would be cool to refactor this to return an Option instead of a string error message, and do the user-view formatting at a higher level with `flatMap` or something to handle the nones